dev/com.ibm.ws.app.manager.war/resources/OSGI-INF/l10n/metatype.properties

```
@@ -31,4 +31,9 @@ classloader.desc=Defines the settings for the application classloader.
 
 appBnd.name=Application Bindings
 appBnd.name$Ref=Application bindings reference
-appBnd.desc=Binds general deployment information included in the application to specific resources.
\ No newline at end of file
+appBnd.desc=Binds general deployment information included in the application to specific resources.
+
+earapp.addWebModuleClassPathTo.name=The class loader used for Web Application class paths
+earapp.addWebModuleClassPathTo.desc=Defines the class loader used for JARs referenced by Web Application Class-Path headers.
+earapp.addWebModuleClassPathTo.warLoader=Use the Web Application class loader for the JARs referenced by a Web Application Class-Path header.
```

```suggestion
earapp.addWebModuleClassPathTo.warLoader=Use the web application class loader for the JARs that are referenced by a web application class-path header.
```

is this a description of this attribute? Does it take a true/false value or specify a particular class loader? Im not clear on what we want to tell the user about this attribiute

---

The `warLoader` and `earLoader` are not attriburtes.  Instead they are possible option values for the `addWebModuleClassPathTo` attribute.  Seeing the metatype XML may help in understanding:
```xml
        <AD name="%earapp.addWebModuleClassPathTo.name" description="%earapp.addWebModuleClassPathTo.desc"
            id="addWebModuleClassPathTo"
            required="false" 
            type="String" 
            ibm:beta="true" >
            <Option label="%earapp.addWebModuleClassPathTo.warLoader" value="warLoader"/>
            <Option label="%earapp.addWebModuleClassPathTo.earLoader" value="earLoader"/>
        </AD>
```
Where the server config would look like this for configuring the `earlLoader` to be used for the `Class-Path` content of a Web Module:

```xml
<enterpriseApplication location="myApp.ear" addWebModuleClassPathTo="earLoader" />
```

Options show up in the configuration reference in the description column.  For example, see the `updateTrigger` attribute which has `disabled`, `mbean` and `polled` as options: https://openliberty.io/docs/latest/reference/config/config.html

With that in mind, my suggested text would look like this in the description column with `warLoader` being in the `Default` column:

> Defines the class loader used for JARs referenced by Web Application Class-Path headers.
> **warLoader**
> Use the Web Application class loader for the JARs referenced by a Web Application Class-Path header.
> **earLoader**
> Use the parent Enterprise Application class loader for the JARs referenced by a Web Application Class-Path header.

And the `Type` column will list `warLoader` and `earLoader`.  I don't mind taking the capitalization out of Web Application and Enterprise Application.  But I think we must keep the header text `Class-Path` as is since that is from the Java JAR specification and we want to make sure we keep the casing there.

---

ok- generally we are asked not to capitalize anything except for stand-alone licensed products. But if `Class-Path` is how it appears in code, thats the way it should appear here.

Thanks for the explanation - makes sense. I wonder if it would be helpful to add something like

> Defines the class loader that is used for JARs that are referenced by web application Class-Path headers. Specify one of the following values....

but im not sure if this description might be used in other contexts where the value descriptions dont appear just after, ie does it need to be able to make sense without them

---

I don't think the attribute description is shown without the options and the option descriptions.